src/Stack/Build/ConstructPlan.hs

```diff
@@ -1028,9 +1029,8 @@ pprintExceptions exceptions stackYaml stackRoot parentMap wanted' prunedGlobalDe
        go (name, (_range, Just (version,cabalHash), DependencyMismatch{})) =
            Map.singleton name (version, cabalHash)
        go _ = Map.empty
-    pprintExtra (name, (version, BlobKey cabalHash cabalSize)) =
-      let cfInfo = CFIHash cabalHash (Just cabalSize)
-          packageIdRev = PackageIdentifierRevision name version cfInfo
+    pprintExtra (name, (version, BlobKey _ _)) =
+      let packageIdRev = PackageIdentifierRevision name version CFILatest
```

Comment:
This is clearly for the suggestions part. Without this, if a package is not in the snapshot - The stack build tool will recommend it to be added like this:

```
  * Recommended action: try adding the following to your extra-deps in /home/sibi/github/tldr-hs2/stack.yaml:

string-quote-0.0.1@sha256:7d91a0ba1be44b2443497c92f2f027cd4580453b893f8b5ebf061e1d85befaf3
```

With the above change, it will recommend it like this:

```
  * Recommended action: try adding the following to your extra-deps in /home/sibi/github/tldr-hs2/stack.yaml:

string-quote-0.0.1
```

The reason I made that change was because of Michael's comment in the [issue](https://github.com/commercialhaskell/stack/issues/4288):

> 
>... the original stack.yaml can be _even more_ lightweight than it is today, e.g. including the `@rev:0` information there will no longer be necessary at all.

Reply:
But why not change the suggestion properly instead? This change above complicates the logic a bit an gives no hints about reasons